pkg/yamlmeta/node.go

```diff
@@ -69,13 +70,26 @@ func (a *Array) SetValue(val interface{}) error {
 }
 
 func (ai *ArrayItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into arrayitem", val)
+	if !(val == nil || isMapOrArrayOrScalar(val)) {
```

Comment:
it would be good i think to move nil check within isMapOrArrayOrScalar. (if conditional looks a bit confusing at first glance.)

Reply:
🤔 ... and it also puts pressure on revealing the intention of the function name: to ensure that `val` is assignable as a value to a node within a document.